src/Filters/Block.php

```diff
@@ -509,6 +509,13 @@ private function getAvailabilitiesBlock($filter, $selectedFilters)
                     }
                 }
             }
+
+            // Hide Available option if the count is the same as In stock, it doesn't make no sense
+            // Product count is a reliable indicator here, because there can never be product IN STOCK that is not AVAILABLE
+            // So if the counts match, it MUST BE the same products
+            if ($availabilityOptions[1]['nbr'] == $availabilityOptions[2]['nbr']) {
+                unset($availabilityOptions[1]);
+            }
```

Comment:
I understand the idea here...  but I find it weird that the filter would display the same elements depending on whether "available" and "in stock" elements have the same list

it kinda breaks the UI consistency to have sometimes both and sometimes "available" disappears...  

but maybe I'm wrong, I'm not a UX expert

has this been validated with the product team?